catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostProcessPlugin.java

```diff
@@ -155,68 +171,102 @@ private boolean isNitfMimeType(String rawMimeType) {
     return false;
   }
 
-  private void process(List<ContentItem> contentItems) {
-    List<ContentItem> newContentItems = new ArrayList<>();
-    contentItems.forEach(contentItem -> process(contentItem, newContentItems));
-    contentItems.addAll(newContentItems);
+  public void setMaxSideLength(int maxSideLength) {
+    if (maxSideLength > 0) {
+      LOGGER.trace("Setting derived image maxSideLength to {}", maxSideLength);
+      this.maxSideLength = maxSideLength;
+    } else {
+      LOGGER.debug(
+              "Invalid `maxSideLength` value [{}], must be greater than zero. Default value [{}] will be used instead.",
+              maxSideLength,
+              DEFAULT_MAX_SIDE_LENGTH);
+      this.maxSideLength = DEFAULT_MAX_SIDE_LENGTH;
+    }
+  }
+
+  public void setMaxNitfSizeMB(int maxNitfSizeMB) {
+    this.maxNitfSizeMB = maxNitfSizeMB;
+  }
+
+  public void setCreateOverview(boolean createOverview) {
+    this.createOverview = createOverview;
+  }
+
+  public void setStoreOriginalImage(boolean storeOriginalImage) {
+    this.storeOriginalImage = storeOriginalImage;
   }
 
-  private void process(ContentItem contentItem, List<ContentItem> contentItems) {
-    Metacard metacard = contentItem.getMetacard();
+  NitfRenderer getNitfRenderer() {
```

Comment:
✏️ Should this be marked `@VisibleForTesting`?

Reply:
Went with @lessarderic's approach